crates/nu-cli/src/repl.rs

```diff
@@ -374,6 +440,17 @@ pub fn evaluate_repl(
         start_time = std::time::Instant::now();
         line_editor = line_editor.with_transient_prompt(transient_prompt);
         let input = line_editor.read_line(&nu_prompt);
+        // we got our inputs, we can now drop our stack references
+        // This lists all of the stack references that we have cleaned up
+        line_editor = line_editor
+            // CLEAR STACK-REFERENCE 1
+            .with_highlighter(Box::<NoOpHighlighter>::default())
+            // CLEAR STACK-REFERENCE 2
+            .with_completer(Box::<DefaultCompleter>::default())
+            // CLEAR STACK-REFERENCE 3
+            .clear_menus()
+            // CLEAR STACK-REFERENCE 4
+            .with_transient_prompt(Box::<DefaultPrompt>::default());
         let shell_integration = config.shell_integration;
```

Comment:
```suggestion
        let shell_integration = config.shell_integration;
        
        unique_stack = Arc::into_inner(stack).expect("only one stack reference");
```
Then, you can unwrap the `Arc` here and assign it back to `unique_stack` for use below.

These changes should eliminate a bunch of `get_unique_mut` (which can panic). Also, it might even make it possible to remove the `SharedStack` (with the exception of the `child_stack` function).